neptune/query/query.go

```diff
@@ -112,11 +112,10 @@ const (
 	CreateObservationPart          = `g.addV('_%s_observation').as('o').property(single, 'value', '%s').property(single, 'rowIndex', '%d')`
 	AddObservationRelationshipPart = `.V().hasId('%s').hasLabel('_%s_%s').where(values('value').is("%s")).addE('isValueOf').from('o')`
 
-	GetInstanceHeaderPart  = `g.V().hasLabel('_%s_Instance').values('header').aggregate('results')`
-	GetAllObservationsPart = `.V().hasLabel('_%s_observation').values('row')`
-
-	GetFirstDimensionPart       = `.V().has('_%s_%s','value',within(%s)).in('isValueOf')`
-	GetObservationDimensionPart = `__.as('row').out('isValueOf').has('_%s_%s','value',within(%s))`
-	GetObservationSelectRowPart = `.select('row').by('value').aggregate('results').cap('results').unfold()`
+	GetInstanceHeaderPart       = `g.V().hasLabel('_%s_Instance').values('header').aggregate('results')`
+	GetAllObservationsPart      = `.V().hasLabel('_%s_observation').values('row')`
+	GetFirstDimensionPart       = `.V().hasId(%s).in('isValueOf')`
+	GetAdditionalDimensionsPart = `.where(out('isValueOf').hasId(%s).fold().count(local).is_(%d))`
+	GetObservationSelectPart    = `.values('value').aggregate('results').cap('results').unfold()`
```

Comment:
Can we possibly reorder this so GetObsercation comes before GetFirstDimension so they're listed in the order they're used? It doesnt matter hugely, and it might just be me that expects them in a certain order, but i had to go back and check the code a few times to try and build the query in my mind frmo these strings to see how things fit together

Reply:
I think this is in the order that they are used? (this was my intention) - the `GetObservationSelectPart` is used at the end of the query to pull out the CSV row value. I realise now that this is badly named. I will update the name.

Reply:
ohhhh sorry, git hid the few lines that mark most of your changes as being in the buildObservation() function so I misunderstodd!